internal/pkg/describe/lb_web_app.go

```diff
@@ -182,48 +189,76 @@ func (d *WebAppDescriber) Describe(shouldOutputResources bool) (*WebApp, error)
 }
 
 // URI returns the WebAppURI to identify this application uniquely given an environment name.
-func (d *WebAppDescriber) URI(envName string) (*WebAppURI, error) {
-	env, err := d.store.GetEnvironment(d.app.Project, envName)
+func (d *WebAppDescriber) URI(envName string) (string, error) {
+	err := d.initAppDescriber(envName)
 	if err != nil {
-		return nil, fmt.Errorf("get environment %s: %w", envName, err)
+		return "", err
 	}
 
-	envOutputs, err := d.stackDescriber.EnvOutputs(env)
+	envOutputs, err := d.appDescriber.EnvOutputs()
 	if err != nil {
-		return nil, fmt.Errorf("get output for environment %s: %w", envName, err)
+		return "", fmt.Errorf("get output for environment %s: %w", envName, err)
 	}
-	appParams, err := d.stackDescriber.AppParams(env, d.app.Name)
+	appParams, err := d.appDescriber.AppParams()
 	if err != nil {
-		return nil, fmt.Errorf("get parameters for application %s: %w", d.app.Name, err)
+		return "", fmt.Errorf("get parameters for application %s: %w", d.app.Name, err)
 	}
 
 	uri := &WebAppURI{
-		DNSName: envOutputs[CFNStack.EnvOutputPublicLoadBalancerDNSName],
-		Path:    appParams[CFNStack.LBWebAppRulePathParamKey],
+		DNSName: envOutputs[stack.EnvOutputPublicLoadBalancerDNSName],
+		Path:    appParams[stack.LBWebAppRulePathParamKey],
 	}
-	_, isHTTPS := envOutputs[CFNStack.EnvOutputSubdomain]
+	_, isHTTPS := envOutputs[stack.EnvOutputSubdomain]
 	if isHTTPS {
-		dnsName := fmt.Sprintf("%s.%s", d.app.Name, envOutputs[CFNStack.EnvOutputSubdomain])
+		dnsName := fmt.Sprintf("%s.%s", d.app.Name, envOutputs[stack.EnvOutputSubdomain])
 		uri = &WebAppURI{
 			DNSName: dnsName,
 		}
 	}
-	return uri, nil
+	return uri.String(), nil
+}
+
+// EnvVars contains serialized environment variables for an application.
+type EnvVars struct {
+	Environment string `json:"environment"`
+	Name        string `json:"name"`
+	Value       string `json:"value"`
+}
+
+// CfnResource contains application resources created by cloudformation.
+type CfnResource struct {
+	Type       string `json:"type"`
+	PhysicalID string `json:"physicalID"`
+}
+
+// WebAppConfig contains serialized configuration parameters for a web application.
+type WebAppConfig struct {
+	Environment string `json:"environment"`
+	Port        string `json:"port"`
+	Tasks       string `json:"tasks"`
+	CPU         string `json:"cpu"`
+	Memory      string `json:"memory"`
+}
+
+// WebAppRoute contains serialized route parameters for a web application.
+type WebAppRoute struct {
```

Comment:
Does the json marshaling not work if the structs are unexported?

Reply:
Yeah, also testing in CLI package uses this struct.